go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -294,7 +293,7 @@ func localnetDeleteService(svc *kapi.Service) error {
 		return err
 	}
 	rules := localnetIptRules(svc)
-	logrus.Debugf("Delete rules %v for service %v", rules, svc.Name)
+	klog.V(5).Infof("Delete rules %v for service %v", rules, svc.Name)
```

Comment:
this is not as intuitive as logrus. I know you said why this has to be done, but just pointing that out.

Reply:
@trozet @alexanderConstantinescu It's worth noting the Kubernetes never calls V(x) with anything other than Infof(). And it uses V(5).Infof() for "trace" level logging, which I think is what Alexander is doing here?

We don't /have/ to use V(5).Infof() a lot if we don't want to, just where we think it should be even more verbose than debug level.

One thing I'd prefer to see here is to keep our existing loglevel config option values (eg 0-5) and map those to klog ourselves. Either that or clarify the help text, because I think it's too confusing right now. Explicitly listing the levels is much clearer. We can always add a 6=trace to the list that duplicates what -V5 does?

Reply:
>  And it uses V(5).Infof() for "trace" level logging, which I think is what Alexander is doing here?

Well not exactly. Today `logrus.Debug` is something not logged by default in ovnkube. We need to explicitly set the level to `dbg` to get those logs. I wanted to keep this functionality, and as our current default log level  was `loglevel = 4`, I decided to set debug logs to `klog.V(5).Info`

> One thing I'd prefer to see here is to keep our existing loglevel config option values (eg 0-5) and map those to klog ourselves. 

Not sure I follow completely, if I understand you correctly @dcbw you'd like to map the old to new levels as follows:

```
logrus.Err -> klog.Err
logrus.Info -> klog.V(1).Info
logrus.Warn -> klog.V(2).Info
logrus.Debug -> klog.V(5).Info
logrus.Trace -> klog.V(6).Info    # Which we never had in the code 
```

Something like that?

Reply:
> `logrus.Err -> klog.Err`
> `logrus.Info -> klog.V(1).Info`

You don't need the "`.V(1)`" in that. It's the default.

> `logrus.Warn -> klog.V(2).Info`

That's backwards; Warn is more severe than Info, not less. Anyway, there's `klog.Warn`

> `logrus.Debug -> klog.V(5).Info`
> `logrus.Trace -> klog.V(6).Info # Which we never had in the code`

I think in k8s/OCP there's somewhat of a standard that processes should run with `-v 2` in production, and anything higher than that is debug-ish. And `V(5)` is a firehose. So maybe, `logrus.Debug` stuff that isn't very verbose can be `klog.V(4).Info` and `logrus.Debug` stuff that is going to fill up your logs very quickly and make it hard to see anything else should be `klog.V(5).Info`

Reply:
> Not sure I follow completely, if I understand you correctly @dcbw you'd like to map the old to new levels as follows:

@alexanderConstantinescu no, I'd like to have warn=warn, info=info, and error=error :) Debug doesn't existing in klog, so V(5).Info is fine for debug. Point being, if you want to warn or error log something, it should *always* be logged (or you wouldn't have made it error/warning level in the first place). So:

```
logrus.Err -> klog.Err
logrus.Info -> klog.Info
logrus.Warn -> klog.Warn
logrus.Debug -> klog.V(5).Info
```

Or and if we need Trace level at some future point we can always add it. But maybe we just keep things simple for now.

Reply:
>So:
> 
> ```
> logrus.Err -> klog.Err
> logrus.Info -> klog.Info
> logrus.Warn -> klog.Warn
> logrus.Debug -> klog.V(5).Info
> ```

This is _exactly_ what this PR implements. I think that's where my confusion came from - I didn't really understand what the change would be w.r.t what I've already done.

Concerning having a trace/firehose level, which you and @danwinship are touching on:

> So maybe, logrus.Debug stuff that isn't very verbose can be klog.V(4).Info and logrus.Debug stuff that is going to fill up your logs very quickly and make it hard to see anything else should be klog.V(5).Info

I wouldn't be against it, but I don't feel comfortable going through all debug logs in ovnkube and deciding on which should be debug vs trace. I honestly feel more comfortable leaving it "as is" with the original log levels each author decided.

For now, this patch implements the change specified at the top of this post. Meaning, functionally things remain the same but with better performance - i.e the default log level remains 4 and we continue logging `info`, `err` and `warn`, and should a user need `debug` they'd just need to increase the level to 5.

Reply:
@alexanderConstantinescu so I'm OK with this mapping, but have comments about how we treat our existing --loglevel in https://github.com/ovn-org/ovn-kubernetes/pull/1047/commits/593b747369c938a2308fd06cf45192da94e07c73#r378550648